### ISSUE TITLE
VST2: Some effects' parameters get reset to the defaults when hiding/showing UI

### ISSUE THREAD

**Issue by dozzzzer** [audacity/audacity#3993]
### Plugin

**Melda** plugins, **Rift** by Minimal Audio, **Deegain**

### Plugin Format

VST2

### Plugin source

https://www.meldaproduction.com/MFreeFXBundle
https://www.minimal.audio/products/rift
https://dotec-audio.com/deegain.html

### Bug description

An effect's parameters get reset to the defaults when hiding/showing UI

### Steps to reproduce

1. have some audio
2. start playback
3. select a Melda plugin in the RT effect panel and open it (e.g. MEqualizer)
4. tweak some parameters
5. close and then open the effect window again
6. observe the parameters

### Expected behavior

The parameters stick

### Actual behavior

The parameters are getting reset to the default:

https://user-images.githubusercontent.com/70063970/204591229-8800eb91-865f-4ce5-b781-e4f183d10595.mp4

### Audacity Version

current master / alpha / nightly build

### Operating system

Windows, macOS

### Architecture

Apple Silicon (M1, M2, ...)

**Comment by Paul-Licameli** [comment]
I succeeded in reproducing this with MEqualizer but not with DeeGain.

I observe that adding the effect before starting play also experiences this problem.

But I also see that if the effect was previously saved to a project file that is reopened -- then the problem does not happen.

There is a clue for further debugging.

**Comment by dozzzzer** [comment]
Testing against #4006 shows the bug is still there.

**Scenario 1: pauses between Play/Stop**

- have some audio
- add Deegain in the sidebar
- open its UI (the fader is at 0.0 dB)
- start playback by pressing Space
- move the fader down to -20.0 while playing the track
- stop the playback by pressing Space
- wait 1 second
- now play the track again for a few seconds
- stop it by pressing Space
- play again by pressing Space

Act.result: the slider is moved back to 0.0 dB

**Scenario 2: Play/Stop is done with no pauses**
- have some audio
- add Deegain in the sidebar
- open its UI (the fader is at 0.0 dB)
- start playback by pressing Space
- move the fader down to -20.0 while playing the track
- hit the Space key **twice** to stop and play the track respectively

Act.result: the slider is moved back to 0.0 dB

**Comment by pietro68** [comment]
Please let's move this Scenario1 & Scenario2 to a new issue. These are new bugs which are unrelated to the original one - no hiding/showing of the UI is done here.

**Comment by Paul-Licameli** [comment]
I have pushed an added fix onto the branch.  It may fix it all.  Building here: https://github.com/audacity/audacity/actions/runs/3596206050

**Comment by dozzzzer** [comment]
>Please let's move this Scenario1 & Scenario2 to a new issue.

The new ticket is #4009.

>I have pushed an added fix onto the branch. It may fix it all.

Tested. The problem described in #4009 is still there.

**Comment by AnitaBats** [comment]
fixed
